layouts/partials/mobile-sidebar.html

```diff
@@ -0,0 +1,106 @@
+<!-- 
+  Copyright (c) 2018 Eclipse Foundation, Inc.
+
+  This program and the accompanying materials are made available under the
+  terms of the Eclipse Public License v. 2.0 which is available at
+  http://www.eclipse.org/legal/epl-2.0.
+
+  Contributors:
+    Christopher Guindon <chris.guindon@eclipse-foundation.org>
+
+  SPDX-License-Identifier: EPL-2.0
+-->
+
+{{ if isset .Site.Menus "sidebar" }}
+
+{{ $currentPage := . }}
+{{ if ne $currentPage.Params.hide_sidebar true }}
+{{ .Scratch.Set "currentSection" "home" }}
+{{ if $currentPage.Section }}
+  {{ .Scratch.Set "currentSection" $currentPage.Section }}
+{{ end }}
+{{$currentSection := .Scratch.Get "currentSection" }}
+
+<!-------------------- Dropdown Start --------------------------------------->
+  <div class="container">
+  	<div class="row">
+  		<div class="col-sm-24 hidden-lg hidden-md visible-sm visible-xs margin-top-10" id="dropdown-container">
+        {{ range .Site.Menus.sidebar }}
+          {{$parentItem := . }}
+          {{/*  {{ if .HasChildren }}  */}}
+          {{ if and (.HasChildren) (eq $currentSection $parentItem.Identifier) }}
```

Comment:
This seems flakey...are we sure that this triggers properly through all sub-sections within a section?

Reply:
Actually I just found I do need to use this logic since that's the same as sidebar is using. Some sidebars we only want to show them on specific pages, and my mobile sidebar needs to be consistent with the desktop sidebar. I have updated my code to better illustrate. (I do have some logic flaw before 🙈 ) But we need this `if and (.HasChildren) (eq $currentSection $parentItem.Identifier)`